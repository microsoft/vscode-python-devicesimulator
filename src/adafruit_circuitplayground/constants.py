# Copyright (c) Microsoft Corporation.
# Licensed under the MIT license.

ASSIGN_PIXEL_TYPE_ERROR = "The pixel color value type should be tuple, list or hexadecimal."

BRIGHTNESS_RANGE_ERROR = "The brightness value should be a number between 0 and 1."

INDEX_ERROR = "The index is not a valid number, you can access the Neopixels from 0 to 9."

NOT_IMPLEMENTED_ERROR = "This method is not implemented by the simulator"

NOT_SUITABLE_FILE_ERROR = "Your .wav file is not suitable for the Circuit Playground Express."

PIXEL_RANGE_ERROR = "The pixel hexadicimal color value should be in range #000000 and #FFFFFF."

VALID_PIXEL_ASSIGN_ERROR = "The pixel color value should be a tuple with three values between 0 and 255 or a hexadecimal color between 0x000000 and 0xFFFFFF."
