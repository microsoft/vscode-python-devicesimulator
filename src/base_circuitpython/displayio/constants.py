PIXEL_OUT_OF_BOUNDS = "pixel coordinates out of bounds"
PALETTE_OUT_OF_RANGE = "Palette index out of range"
TILE_OUT_OF_BOUNDS = "Tile index out of bounds"
INCORR_SUBCLASS = "Layer must be a Group or TileGrid subclass."
LAYER_ALREADY_IN_GROUP = "Layer already in a group."
GROUP_FULL = "Group Full"
SCALE_TOO_SMALL = "scale must be >= 1"

SCREEN_HEIGHT_WIDTH = 240

CLUE = "CLUE"
BASE_64 = "display_base64"
