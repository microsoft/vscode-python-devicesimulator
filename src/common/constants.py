MAC_OS = "darwin"

TIME_DELAY = 0.03
