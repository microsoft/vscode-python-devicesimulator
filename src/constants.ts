// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import * as path from "path";
import { MessageItem } from "vscode";
import * as nls from "vscode-nls";

// Debugger Server
export const SERVER_INFO = {
  DEFAULT_SERVER_PORT: 5577,
  ERROR_CODE_INIT_SERVER: "ERROR_INIT_SERVER",
  SERVER_PORT_CONFIGURATION: "deviceSimulatorExpress.debuggerServerPort"
};

const localize: nls.LocalizeFunc = nls.config({
  messageFormat: nls.MessageFormat.file
})();

export const CONFIG = {
  SHOW_DEPENDENCY_INSTALL: "deviceSimulatorExpress.showDependencyInstall",
  SHOW_NEW_FILE_POPUP: "deviceSimulatorExpress.showNewFilePopup"
};

export const CONSTANTS = {
  DEBUG_CONFIGURATION_TYPE: "deviceSimulatorExpress",
  DEPENDENCY_CHECKER: {
    PIP3: "pip3",
    PYTHON: "python",
    PYTHON3: "python3.7",
  },
  ERROR: {
    COMPORT_UNKNOWN_ERROR:
      "Writing to COM port (GetOverlappedResult): Unknown error code 121",
    CPX_FILE_ERROR: localize(
      "error.cpxFileFormat",
      "The cpx.json file format is not correct."
    ),
    DEBUGGER_SERVER_INIT_FAILED: (port: number) => {
      return localize(
        "error.debuggerServerInitFailed",
        `Warning : The Debugger Server cannot be opened. Please try to free the port ${port} if it's already in use or select another one in your Settings 'Device Simulator Express: Debugger Server Port' and start another debug session.\n You can still debug your code but you won't be able to use the Simulator.`
      );
    },
    DEBUGGING_SESSION_IN_PROGESS: localize(
      "error.debuggingSessionInProgress",
      "[ERROR] A debugging session is currently in progress, please stop it before running your code. \n"
    ),
    DEPENDENCY_DOWNLOAD_ERROR:
      "Package downloads failed. Some functionalities may not work. Try restarting the simulator or review the installation docs.",
    FAILED_TO_OPEN_SERIAL_PORT: (port: string): string => {
      return localize(
        "error.failedToOpenSerialPort",
        `[ERROR] Failed to open serial port ${port}.`
      );
    },
    FAILED_TO_OPEN_SERIAL_PORT_DUE_TO: (port: string, error: any) => {
      return localize(
        "error.failedToOpenSerialPortDueTo",
        `[ERROR] Failed to open serial port ${port} due to error: ${error}. \n`
      );
    },
    INCORRECT_FILE_NAME_FOR_DEVICE: localize(
      "error.incorrectFileNameForDevice",
      '[ERROR] Can\'t deploy to your Circuit Playground Express device, please rename your file to "code.py" or "main.py". \n'
    ),
    INCORRECT_FILE_NAME_FOR_DEVICE_POPUP: localize(
      "error.incorrectFileNameForDevicePopup",
      'Seems like you have a different file name than what CPX requires, please rename it to "code.py" or "main.py".'
    ),
    INVALID_FILE_EXTENSION_DEBUG: localize(
      "error.invalidFileExtensionDebug",
      "The file you tried to run isn't a Python file."
    ),
    NO_DEVICE: localize(
      "error.noDevice",
      "No plugged in boards detected. Please double check if your board is connected and/or properly formatted"
    ),
    NO_FILE_TO_RUN: localize(
      "error.noFileToRun",
      '[ERROR] We can\'t find a Python file to run. Please make sure you select or open a new ".py" code file, or use the "New File" command to get started and see useful links.\n'
    ),
    NO_FOLDER_OPENED: localize(
      "error.noFolderCreated",
      "In order to use the Serial Monitor, you need to open a folder and reload VS Code."
    ),
    NO_PROGRAM_FOUND_DEBUG: localize(
      "error.noProgramFoundDebug",
      "Cannot find a program to debug."
    ),
    NO_PYTHON_PATH: localize(
      "error.noPythonPath",
      "We found that you don't have Python 3 installed on your computer, please install the latest version, add it to your PATH and try again."
    ),
    RECONNECT_DEVICE: localize(
      "error.reconnectDevice",
      "Please disconnect your Circuit Playground Express and try again."
    ),
    STDERR: (data: string) => {
      return localize("error.stderr", `\n[ERROR] ${data} \n`);
    },
    UNEXPECTED_MESSAGE: localize(
      "error.unexpectedMessage",
      "Webview sent an unexpected message"
    )
  },
  FILESYSTEM: {
    OUTPUT_DIRECTORY: "out",
    PYTHON_LIBS_DIR: "python_libs"
  },
  INFO: {
    ARE_YOU_SURE: localize(
      "info.areYouSure",
      "Are you sure you don't want to install the dependencies? The extension can't run without installing it"
    ),
    CLOSED_SERIAL_PORT: (port: string) => {
      return localize(
        "info.closedSerialPort",
        `[DONE] Closed the serial port - ${port} \n`
      );
    },
    COMPLETED_MESSAGE: "Completed",
    CPX_JSON_ALREADY_GENERATED: localize(
      "info.cpxJsonAlreadyGenerated",
      "cpx.json has already been generated."
    ),
    DEPLOY_DEVICE: localize(
      "info.deployDevice",
      "\n[INFO] Deploying code to the device...\n"
    ),
    DEPLOY_SIMULATOR: localize(
      "info.deploySimulator",
      "\n[INFO] Deploying code to the simulator...\n"
    ),
    DEPLOY_SUCCESS: localize(
      "info.deploySuccess",
      "\n[INFO] Code successfully copied! Your Circuit Playground Express should be loading and ready to go shortly.\n"
    ),
    EXTENSION_ACTIVATED: localize(
      "info.extensionActivated",
      "Congratulations, your extension Adafruit_Simulator is now active!"
    ),
    FILE_SELECTED: (filePath: string) => {
      return localize(
        "info.fileSelected",
        `[INFO] File selected : ${filePath} \n`
      );
    },
    FIRST_TIME_WEBVIEW: localize(
      "info.firstTimeWebview",
      'To reopen the simulator click on the "Open Simulator" button on the upper right corner of the text editor, or select the command "Open Simulator" from command palette.'
    ),
    INCORRECT_FILE_NAME_FOR_SIMULATOR_POPUP: localize(
      "info.incorrectFileNameForSimulatorPopup",
      'We want your code to work on your actual board as well. Make sure you name your file "code.py" or "main.py" to be able to run your code on an actual physical device'
    ),
    INSTALLING_PYTHON_DEPENDENCIES: localize(
      "info.installingPythonDependencies",
      "The Python packages are currently being installed. You will be prompt a message telling you when the installation is done."
    ),
    INSTALL_PYTHON_DEPENDENCIES: localize(
      "info.installPythonDependencies",
      "Do you want us to try and install this extensions dependencies for you?"
    ),
    INVALID_FILE_NAME_DEBUG: localize(
      "info.invalidFileNameDebug",
      'The file you tried to debug isn\'t named "code.py" or "main.py". Rename your file if you want your code to work on your actual device.'
    ),
    NEW_FILE: localize(
      "info.newFile",
      "New to Python or the Circuit Playground Express? We are here to help!"
    ),
    OPENED_SERIAL_PORT: (port: string) => {
      return localize(
        "info.openedSerialPort",
        `[INFO] Opened the serial port - ${port} \n`
      );
    },
    OPENING_SERIAL_PORT: (port: string) => {
      return localize(
        "info.openingSerialPort",
        `[STARTING] Opening the serial port - ${port} \n`
      );
    },
    PLEASE_OPEN_FOLDER: localize(
      "info.pleaseOpenFolder",
      "Please open a folder first."
    ),
    REDIRECT: localize("info.redirect", "You are being redirected."),
    RUNNING_CODE: localize("info.runningCode", "Running user code"),
    SUCCESSFUL_INSTALL: localize(
      "info.successfulInstall",
      "Successfully installed Python dependencies."
    ),
    THIRD_PARTY_WEBSITE: localize(
      "info.thirdPartyWebsite",
      'By clicking "Agree and Proceed" you will be redirected to adafruit.com, a third party website not managed by Microsoft. Please note that your activity on adafruit.com is subject to Adafruit\'s privacy policy'
    ),
    WELCOME_OUTPUT_TAB: localize(
      "info.welcomeOutputTab",
      "Welcome to the Adafruit Simulator output tab!\n\n"
    )
  },
  LABEL: {
    WEBVIEW_PANEL: localize("label.webviewPanel", "Device Simulator Express")
  },
  LINKS: {
    DOWNLOAD_PYTHON: "https://www.python.org/downloads/",
    EXAMPLE_CODE:
      "https://github.com/adafruit/Adafruit_CircuitPython_CircuitPlayground/tree/master/examples",
    HELP:
      "https://learn.adafruit.com/adafruit-circuit-playground-express/circuitpython-quickstart",
    INSTALL: "https://github.com/microsoft/vscode-python-devicesimulator/blob/dev/docs/install.md",
    PRIVACY: "https://www.adafruit.com/privacy",
    TUTORIALS:
      "https://learn.adafruit.com/circuitpython-made-easy-on-circuit-playground-express/circuit-playground-express-library"
  },
  MISC: {
    SELECT_PORT_PLACEHOLDER: localize(
      "misc.selectPortPlaceholder",
      "Select a serial port"
    ),
    SERIAL_MONITOR_NAME: localize(
      "misc.serialMonitorName",
      "Device Simulator Express Serial Monitor"
    ),
    SERIAL_MONITOR_TEST_IF_OPEN: localize(
      "misc.testIfPortOpen",
      "Test if serial port is open"
    )
  },
  NAME: localize("name", "Device Simulator Express"),
  WARNING: {
    ACCEPT_AND_RUN: localize(
      "warning.agreeAndRun",
      "By selecting ‘Agree and Run’, you understand the extension executes Python code on your local computer, which may be a potential security risk."
    ),
    INVALID_BAUD_RATE: localize(
      "warning.invalidBaudRate",
      "Invalid baud rate, keep baud rate unchanged."
    ),
    NO_RATE_SELECTED: localize(
      "warning.noRateSelected",
      "No rate is selected, keep baud rate unchanged."
    ),
    NO_SERIAL_PORT_SELECTED: localize(
      "warning.noSerialPortSelected",
      "No serial port was selected, please select a serial port first"
    ),
    SERIAL_MONITOR_ALREADY_OPENED: (port: string) => {
      return localize(
        "warning.serialMonitorAlreadyOpened",
        `Serial monitor is already opened for ${port} \n`
      );
    },
    SERIAL_MONITOR_NOT_STARTED: localize(
      "warning.serialMonitorNotStarted",
      "Serial monitor has not been started."
    ),
    SERIAL_PORT_NOT_STARTED: localize(
      "warning.serialPortNotStarted",
      "Serial port has not been started.\n"
    )
  }
};

export enum CONFIG_KEYS {
  ENABLE_USB_DETECTION = "deviceSimulatorExpress.enableUSBDetection"
}

export enum TelemetryEventName {
  FAILED_TO_OPEN_SIMULATOR = "SIMULATOR.FAILED_TO_OPEN",

  // Extension commands
  COMMAND_DEPLOY_DEVICE = "COMMAND.DEPLOY.DEVICE",
  COMMAND_NEW_FILE = "COMMAND.NEW.FILE",
  COMMAND_OPEN_SIMULATOR = "COMMAND.OPEN.SIMULATOR",
  COMMAND_RUN_SIMULATOR_BUTTON = "COMMAND.RUN.SIMULATOR_BUTTON",
  COMMAND_RUN_PALETTE = "COMMAND.RUN.PALETTE",
  COMMAND_RUN_EDITOR_ICON = "COMMAND.RUN.EDITOR_ICON",
  COMMAND_SERIAL_MONITOR_CHOOSE_PORT = "COMMAND.SERIAL_MONITOR.CHOOSE_PORT",
  COMMAND_SERIAL_MONITOR_OPEN = "COMMAND.SERIAL_MONITOR.OPEN",
  COMMAND_SERIAL_MONITOR_BAUD_RATE = "COMMAND.SERIAL_MONITOR.BAUD_RATE",
  COMMAND_SERIAL_MONITOR_CLOSE = "COMMAND.SERIAL_MONITOR.CLOSE",

  // Simulator interaction
  SIMULATOR_BUTTON_A = "SIMULATOR.BUTTON.A",
  SIMULATOR_BUTTON_B = "SIMULATOR.BUTTON.B",
  SIMULATOR_BUTTON_AB = "SIMULATOR.BUTTON.AB",
  SIMULATOR_SWITCH = "SIMULATOR.SWITCH",

  // Sensors
  SIMULATOR_TEMPERATURE_SENSOR = "SIMULATOR.TEMPERATURE",
  SIMULATOR_LIGHT_SENSOR = " SIMULATOR.LIGHT",
  SIMULATOR_MOTION_SENSOR = "SIMULATOR.MOTION",
  SIMULATOR_SHAKE = "SIMULATOR.SHAKE",
  SIMULATOR_CAPACITIVE_TOUCH = "SIMULATOR.CAPACITIVE.TOUCH",

  // Pop-up dialog
  CLICK_DIALOG_DONT_SHOW = "CLICK.DIALOG.DONT.SHOW",
  CLICK_DIALOG_EXAMPLE_CODE = "CLICK.DIALOG.EXAMPLE.CODE",
  CLICK_DIALOG_HELP_DEPLOY_TO_DEVICE = "CLICK.DIALOG.HELP.DEPLOY.TO.DEVICE",
  CLICK_DIALOG_TUTORIALS = "CLICK.DIALOG.TUTORIALS",

  ERROR_PYTHON_DEVICE_PROCESS = "ERROR.PYTHON.DEVICE.PROCESS",
  ERROR_PYTHON_PROCESS = "ERROR.PYTHON.PROCESS",
  ERROR_COMMAND_NEW_FILE = "ERROR.COMMAND.NEW.FILE",
  ERROR_DEPLOY_WITHOUT_DEVICE = "ERROR.DEPLOY.WITHOUT.DEVICE",

  SUCCESS_COMMAND_DEPLOY_DEVICE = "SUCCESS.COMMAND.DEPLOY.DEVICE",

  // Performance
  PERFORMANCE_DEPLOY_DEVICE = "PERFORMANCE.DEPLOY.DEVICE",
  PERFORMANCE_NEW_FILE = "PERFORMANCE.NEW.FILE",
  PERFORMANCE_OPEN_SIMULATOR = "PERFORMANCE.OPEN.SIMULATOR"
}

export enum WebviewMessages {
  BUTTON_PRESS = "button-press",
  PLAY_SIMULATOR = "play-simulator",
  SENSOR_CHANGED = "sensor-changed",
  REFRESH_SIMULATOR = "refresh-simulator",
  SLIDER_TELEMETRY = "slider-telemetry"
}

// tslint:disable-next-line: no-namespace
export namespace DialogResponses {
  export const ACCEPT_AND_RUN: MessageItem = {
    title: localize("dialogResponses.agreeAndRun", "Agree and Run")
  };
  export const AGREE_AND_PROCEED: MessageItem = {
    title: localize("dialogResponses.agreeAndProceed", "Agree and Proceed")
  };
  export const CANCEL: MessageItem = {
    title: localize("dialogResponses.cancel", "Cancel")
  };
  export const HELP: MessageItem = {
    title: localize("dialogResponses.help", "I need help")
  };
  export const DONT_SHOW: MessageItem = {
    title: localize("dialogResponses.dontShowAgain", "Don't Show Again")
  };
  export const NO: MessageItem = {
    title: localize("dialogResponses.No", "No")
  };
  export const INSTALL_NOW: MessageItem = {
    title: localize("dialogResponses.installNow", "Install Now")
  };
  export const DONT_INSTALL: MessageItem = {
    title: localize("dialogResponses.dontInstall", "Don't Install")
  };
  export const PRIVACY_STATEMENT: MessageItem = {
    title: localize("info.privacyStatement", "Privacy Statement")
  };
  export const TUTORIALS: MessageItem = {
    title: localize("dialogResponses.tutorials", "Tutorials on Adafruit")
  };
  export const EXAMPLE_CODE: MessageItem = {
    title: localize("dialogResponses.exampleCode", "Example Code on GitHub")
  };
  export const MESSAGE_UNDERSTOOD: MessageItem = {
    title: localize("dialogResponses.messageUnderstood", "Got It")
  };
  export const INSTALL_PYTHON: MessageItem = {
    title: localize("dialogResponses.installPython", "Install from python.org")
  };
  export const YES: MessageItem = {
    title: localize("dialogResponses.Yes", "Yes")
  };
  export const READ_INSTALL_MD: MessageItem = {
    title: localize("dialogResponses.readInstall", "Read installation docs")
  };
}

export const CPX_CONFIG_FILE = path.join(".vscode", "cpx.json");

export const USER_CODE_NAMES = {
  CODE_PY: "code.py",
  MAIN_PY: "main.py"
};

export const STATUS_BAR_PRIORITY = {
  PORT: 20,
  OPEN_PORT: 30,
  BAUD_RATE: 40
};

export default CONSTANTS;
