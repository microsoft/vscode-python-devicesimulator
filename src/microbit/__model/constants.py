MICROBIT = "micro:bit"

# string arguments for constructor
BLANK_5X5 = "00000:00000:00000:00000:00000:"

# pre-defined image patterns
IMAGE_PATTERNS = {
    "HEART": "09090:99999:99999:09990:00900:",
    "HEART_SMALL": "00000:09090:09990:00900:00000:",
    "HAPPY": "00000:09090:00000:90009:09990:",
    "SMILE": "00000:00000:00000:90009:09990:",
    "SAD": "00000:09090:00000:09990:90009:",
    "CONFUSED": "00000:09090:00000:09090:90909:",
    "ANGRY": "90009:09090:00000:99999:90909:",
    "ASLEEP": "00000:99099:00000:09990:00000:",
    "SURPRISED": "09090:00000:00900:09090:00900:",
    "SILLY": "90009:00000:99999:00909:00999:",
    "FABULOUS": "99999:99099:00000:09090:09990:",
    "MEH": "09090:00000:00090:00900:09000:",
    "YES": "00000:00009:00090:90900:09000:",
    "NO": "90009:09090:00900:09090:90009:",
    "CLOCK12": "00900:00900:00900:00000:00000:",
    "CLOCK11": "09000:09000:00900:00000:00000:",
    "CLOCK10": "00000:99000:00900:00000:00000:",
    "CLOCK9": "00000:00000:99900:00000:00000:",
    "CLOCK8": "00000:00000:00900:99000:00000:",
    "CLOCK7": "00000:00000:00900:09000:09000:",
    "CLOCK6": "00000:00000:00900:00900:00900:",
    "CLOCK5": "00000:00000:00900:00090:00090:",
    "CLOCK4": "00000:00000:00900:00099:00000:",
    "CLOCK3": "00000:00000:00999:00000:00000:",
    "CLOCK2": "00000:00099:00900:00000:00000:",
    "CLOCK1": "00090:00090:00900:00000:00000:",
    "ARROW_N": "00900:09990:90909:00900:00900:",
    "ARROW_NE": "00999:00099:00909:09000:90000:",
    "ARROW_E": "00900:00090:99999:00090:00900:",
    "ARROW_SE": "90000:09000:00909:00099:00999:",
    "ARROW_S": "00900:00900:90909:09990:00900:",
    "ARROW_SW": "00009:00090:90900:99000:99900:",
    "ARROW_W": "00900:09000:99999:09000:00900:",
    "ARROW_NW": "99900:99000:90900:00090:00009:",
    "TRIANGLE": "00000:00900:09090:99999:00000:",
    "TRIANGLE_LEFT": "90000:99000:90900:90090:99999:",
    "CHESSBOARD": "09090:90909:09090:90909:09090:",
    "DIAMOND": "00900:09090:90009:09090:00900:",
    "DIAMOND_SMALL": "00000:00900:09090:00900:00000:",
    "SQUARE": "99999:90009:90009:90009:99999:",
    "SQUARE_SMALL": "00000:09990:09090:09990:00000:",
    "RABBIT": "90900:90900:99990:99090:99990:",
    "COW": "90009:90009:99999:09990:00900:",
    "MUSIC_CROTCHET": "00900:00900:00900:99900:99900:",
    "MUSIC_QUAVER": "00900:00990:00909:99900:99900:",
    "MUSIC_QUAVERS": "09999:09009:09009:99099:99099:",
    "PITCHFORK": "90909:90909:99999:00900:00900:",
    "XMAS": "00900:09990:00900:09990:99999:",
    "PACMAN": "09999:99090:99900:99990:09999:",
    "TARGET": "00900:09990:99099:09990:00900:",
    "TSHIRT": "99099:99999:09990:09990:09990:",
    "ROLLERSKATE": "00099:00099:99999:99999:09090:",
    "DUCK": "09900:99900:09999:09990:00000:",
    "HOUSE": "00900:09990:99999:09990:09090:",
    "TORTOISE": "00000:09990:99999:09090:00000:",
    "BUTTERFLY": "99099:99999:00900:99999:99099:",
    "STICKFIGURE": "00900:99999:00900:09090:90009:",
    "GHOST": "99999:90909:99999:99999:90909:",
    "SWORD": "00900:00900:00900:09990:00900:",
    "GIRAFFE": "99000:09000:09000:09990:09090:",
    "SKULL": "09990:90909:99999:09990:09990:",
    "UMBRELLA": "09990:99999:00900:90900:09900:",
    "SNAKE": "99000:99099:09090:09990:00000:",
}

IMAGE_TUPLE_LOOKUP = {
    "ALL_CLOCKS": [
        "CLOCK12",
        "CLOCK11",
        "CLOCK10",
        "CLOCK9",
        "CLOCK8",
        "CLOCK7",
        "CLOCK6",
        "CLOCK5",
        "CLOCK4",
        "CLOCK3",
        "CLOCK2",
        "CLOCK1",
    ],
    "ALL_ARROWS": [
        "ARROW_N",
        "ARROW_NE",
        "ARROW_E",
        "ARROW_SE",
        "ARROW_S",
        "ARROW_SW",
        "ARROW_W",
        "ARROW_NW",
    ],
}

# 5x5 Alphabet
# Taken from https://raw.githubusercontent.com/micropython/micropython/264d80c84e034541bd6e4b461bfece4443ffd0ac/ports/nrf/boards/microbit/modules/microbitfont.h
ALPHABET = b"\x00\x00\x00\x00\x00\x08\x08\x08\x00\x08\x0a\x4a\x40\x00\x00\x0a\x5f\xea\x5f\xea\x0e\xd9\x2e\xd3\x6e\x19\x32\x44\x89\x33\x0c\x92\x4c\x92\x4d\x08\x08\x00\x00\x00\x04\x88\x08\x08\x04\x08\x04\x84\x84\x88\x00\x0a\x44\x8a\x40\x00\x04\x8e\xc4\x80\x00\x00\x00\x04\x88\x00\x00\x0e\xc0\x00\x00\x00\x00\x08\x00\x01\x22\x44\x88\x10\x0c\x92\x52\x52\x4c\x04\x8c\x84\x84\x8e\x1c\x82\x4c\x90\x1e\x1e\xc2\x44\x92\x4c\x06\xca\x52\x5f\xe2\x1f\xf0\x1e\xc1\x3e\x02\x44\x8e\xd1\x2e\x1f\xe2\x44\x88\x10\x0e\xd1\x2e\xd1\x2e\x0e\xd1\x2e\xc4\x88\x00\x08\x00\x08\x00\x00\x04\x80\x04\x88\x02\x44\x88\x04\x82\x00\x0e\xc0\x0e\xc0\x08\x04\x82\x44\x88\x0e\xd1\x26\xc0\x04\x0e\xd1\x35\xb3\x6c\x0c\x92\x5e\xd2\x52\x1c\x92\x5c\x92\x5c\x0e\xd0\x10\x10\x0e\x1c\x92\x52\x52\x5c\x1e\xd0\x1c\x90\x1e\x1e\xd0\x1c\x90\x10\x0e\xd0\x13\x71\x2e\x12\x52\x5e\xd2\x52\x1c\x88\x08\x08\x1c\x1f\xe2\x42\x52\x4c\x12\x54\x98\x14\x92\x10\x10\x10\x10\x1e\x11\x3b\x75\xb1\x31\x11\x39\x35\xb3\x71\x0c\x92\x52\x52\x4c\x1c\x92\x5c\x90\x10\x0c\x92\x52\x4c\x86\x1c\x92\x5c\x92\x51\x0e\xd0\x0c\x82\x5c\x1f\xe4\x84\x84\x84\x12\x52\x52\x52\x4c\x11\x31\x31\x2a\x44\x11\x31\x35\xbb\x71\x12\x52\x4c\x92\x52\x11\x2a\x44\x84\x84\x1e\xc4\x88\x10\x1e\x0e\xc8\x08\x08\x0e\x10\x08\x04\x82\x41\x0e\xc2\x42\x42\x4e\x04\x8a\x40\x00\x00\x00\x00\x00\x00\x1f\x08\x04\x80\x00\x00\x00\x0e\xd2\x52\x4f\x10\x10\x1c\x92\x5c\x00\x0e\xd0\x10\x0e\x02\x42\x4e\xd2\x4e\x0c\x92\x5c\x90\x0e\x06\xc8\x1c\x88\x08\x0e\xd2\x4e\xc2\x4c\x10\x10\x1c\x92\x52\x08\x00\x08\x08\x08\x02\x40\x02\x42\x4c\x10\x14\x98\x14\x92\x08\x08\x08\x08\x06\x00\x1b\x75\xb1\x31\x00\x1c\x92\x52\x52\x00\x0c\x92\x52\x4c\x00\x1c\x92\x5c\x90\x00\x0e\xd2\x4e\xc2\x00\x0e\xd0\x10\x10\x00\x06\xc8\x04\x98\x08\x08\x0e\xc8\x07\x00\x12\x52\x52\x4f\x00\x11\x31\x2a\x44\x00\x11\x31\x35\xbb\x00\x12\x4c\x8c\x92\x00\x11\x2a\x44\x98\x00\x1e\xc4\x88\x1e\x06\xc4\x8c\x84\x86\x08\x08\x08\x08\x08\x18\x08\x0c\x88\x18\x00\x00\x0c\x83\x60"
# We support ASCII characters between these indexes on the microbit
ASCII_START = 32
ASCII_END = 126
SPACE_BETWEEN_LETTERS_WIDTH = 1
WHITESPACE_WIDTH = 3

# numerical LED values
LED_HEIGHT = 5
LED_WIDTH = 5
BRIGHTNESS_MIN = 0
BRIGHTNESS_MAX = 9

# error messages
BRIGHTNESS_ERR = "brightness out of bounds"
COPY_ERR_MESSAGE = "please call copy function first"
INCORR_IMAGE_SIZE = "image data is incorrect size"
INDEX_ERR = "index out of bounds"
NOT_IMPLEMENTED_ERROR = "This method is not implemented by the simulator"
UNSUPPORTED_ADD_TYPE = "unsupported types for __add__:"
SAME_SIZE_ERR = "images must be the same size"

TIME_DELAY = 0.03
