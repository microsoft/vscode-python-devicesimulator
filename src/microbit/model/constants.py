# string arguments for constructor
BLANK_5X5 = "00000:00000:00000:00000:00000:"
BOAT = "05050:05050:05050:99999:09990:"
HEART = "09090:99999:99999:09990:00900:"

# numerical max values
LED_HEIGHT = 5
LED_WIDTH = 5
BRIGHTNESS_MIN = 0
BRIGHTNESS_MAX = 9

# error messages
BRIGHTNESS_ERR = "brightness out of bounds"
COPY_ERR_MESSAGE = "please copy() first"
INCORR_IMAGE_SIZE = "image data is incorrect size"
INDEX_ERR = "index out of bounds"
NOT_IMPLEMENTED_ERROR = "This method is not implemented by the simulator"
UNSUPPORTED_ADD_TYPE = "unsupported types for __add__:"
SAME_SIZE_ERR = "images must be the same size"
