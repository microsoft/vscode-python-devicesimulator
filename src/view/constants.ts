// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.a

// Key events
export const CONSTANTS = {
  KEYBOARD_KEYS: {
    ENTER: "Enter"
  }
}

export default CONSTANTS;