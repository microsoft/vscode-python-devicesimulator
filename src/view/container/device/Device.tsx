// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import * as React from "react";
import { Cpx } from "../../components/cpx/Cpx";
import { Microbit } from "../../components/microbit/Microbit";
import { DEVICE_LIST_KEY } from "../../constants";

interface IProps {
    currentSelectedDevice: string;
}
// Container to switch between multiple devices

export class Device extends React.Component<IProps, {}> {
    constructor(props: IProps) {
        super(props);
    }
    render() {
        const { currentSelectedDevice } = this.props;

        return (
            <div className="device-container">
                <React.Fragment>
                    {loadSelectedDevice(currentSelectedDevice)}
                </React.Fragment>
            </div>
        );
    }
}
const loadSelectedDevice = (currentSelectedDevice: string) => {
    switch (currentSelectedDevice) {
        case DEVICE_LIST_KEY.CPX:
            return <Cpx />;
        case DEVICE_LIST_KEY.MICROBIT:
            return <Microbit />;
        default:
            return null;
    }
};
