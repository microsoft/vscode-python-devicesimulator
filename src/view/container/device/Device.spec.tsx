import * as React from "react";
import * as ReactDOM from "react-dom";
import { IntlProvider } from "react-intl";
import * as testRenderer from "react-test-renderer";
import { DEVICE_LIST_KEY } from "../../constants";
import { Device } from "./Device";

describe("Device component ", () => {
    it("should render correctly", () => {
        const component = testRenderer
            .create(
                <IntlProvider locale="en">
                    <Device currentSelectedDevice={DEVICE_LIST_KEY.MICROBIT} />
                </IntlProvider>
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it("should render without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <IntlProvider locale="en">
                <Device currentSelectedDevice={DEVICE_LIST_KEY.MICROBIT} />
            </IntlProvider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });
});
