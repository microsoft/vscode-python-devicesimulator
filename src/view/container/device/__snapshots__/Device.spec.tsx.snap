// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device component  should render correctly 1`] = `
<div
  className="device-container"
>
  <p>
    Microbit to be implemented!
  </p>
</div>
`;
